pkg/cluster/mpi/configmap.go

```diff
@@ -14,12 +15,47 @@ import (
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
-var (
-	sshdConfigTmpl = `
-Port %d
-AuthorizedKeysFile %s
-`
-)
+const launchScriptTemplate = `#!/bin/bash
+
+set -o nounset 
+set -o errexit
+
+SSH_PORT=%[1]d
+DOMINO_UID=%[2]d
+DOMINO_GID=%[3]d
+DOMINO_USER=%[4]s
+DOMINO_GROUP=%[5]s
+AUTHORIZED_KEYS_PATH=%[6]s
+
+if ! id $DOMINO_UID >/dev/null 2>&1; then
+    groupadd -g $DOMINO_GID $DOMINO_GROUP
+    useradd -u $DOMINO_UID -g $DOMINO_GID -mN -s /bin/bash $DOMINO_USER
+fi
+if [ "$(id -nu $DOMINO_UID)" != "$DOMINO_USER" ]; then
+    echo >&2 "User name mismatch"
+    exit 1
+fi
+
+DOMINO_HOME=$(eval echo "~$DOMINO_USER")
+SSH_DIR="$DOMINO_HOME/.ssh"
+mkdir -p "$SSH_DIR"
+
+ssh-keygen -f "$SSH_DIR/ssh_host_key" -N '' -t rsa
+chmod 400 "$SSH_DIR/ssh_host_key"
+
+cat << EOF > "$SSH_DIR/sshd_config"
```

Comment:
Is there a reason we moved from mounting this in a configmap? It seems like this could still be mounted statically and this script simply configures the user and then launches `sshd` with `su`. I don't think we actually need the user's home directory, we can just place this wherever and pass that to `sshd` on launch.

Reply:
Yeah, we can... I just thought it's simpler this way: everything is one place, single mount etc. For the price of an ugly `cat << EOF` construct, which doesn't look too terrible here.

Reply:
I guess to me the harness script is a bit of an anti pattern and we should keep it as small as possible. If we don't need any information from the live container to generate the config file, I think is should be mounted read only from a configmap